content/nginx-one/k8s/add-ngf.md

```diff
@@ -0,0 +1,180 @@
+---
+title: Connect NGINX Gateway Fabric
+toc: true
+weight: 300
+nd-content-type: how-to
+nd-product: NGINX One
+---
+
+This document explains how to connect F5 NGINX Gateway Fabric to F5 NGINX One Console using NGINX Agent.
+Connecting NGINX Gateway Fabric to NGINX One Console enables centralized monitoring of all controller instances.
+
+Once connected, you'll see a **read-only** configuration of NGINX Gateway Fabric. For each instance, you can review:
+
+- Read-only configuration file
+- Unmanaged SSL/TLS certificates for Control Planes
+
+## Before you begin
+
+Log in to NGINX One Console. If you need more information, review our [Get started guide]({{< ref "/nginx-one/getting-started.md#before-you-begin" >}}).
+
+You also need:
+
+- Administrator access to a Kubernetes cluster.
+- If you use [Helm](https://helm.sh) and [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl), install them locally.
+
+
+### Create a data plane key
+
+Data plane keys are displayed only once, when you create that key, and cannot be retrieved later.
+
+If you've created and recorded one or more data plane keys, you can edit or revoke those keys. To do so, select **Manage > Data Plane Keys**. NGINX One Console does not store your actual data plane key.
+
+If you've forgotten your data plane key, you can create a new one. Select **Manage > Data Plane Keys > Add Data Plane Key**.
+
+For more options associated with data plane keys, see [Create and manage data plane keys]({{< ref "/nginx-one/connect-instances/create-manage-data-plane-keys.md" >}}).
+
+<!-- Helm tab -->
+
+### Create a Kubernetes secret with the data plane key
+
+To create a Kubernetes secret, you'll need:
+
+- The Data Plane Key
+- To set up the secret in the same namespace as NGINX Gateway Fabric
+- Use the name `dataplane.key` as shown
+- A namespace. The default NGINX Gateway Fabric namespace is `nginx-gateway`
+
+Once you have that information, run the following command:
+
+
+   ```shell
+   kubectl create secret generic dataplane-key \
+     --from-literal=dataplane.key=<Your Dataplane Key> \
+     -n <namespace>
+   ```
+
+## Install Gateway API resources
+<!-- Corresponds to step 2 in the UX -->
+{{< include "/ngf/installation/install-gateway-api-resources.md" >}}
+
+## Install from the OCI registry
+<!-- Corresponds to step 3 in the UX -->
+
+The following steps install NGINX Gateway Fabric directly from the OCI helm registry. If you prefer, you can [install from sources](#install-from-sources) instead.
+
+{{<tabs name="install-helm-oci">}}
+
+{{%tab name="NGINX"%}}
+
+To install the latest stable release of NGINX Gateway Fabric in the **nginx-gateway** namespace, run the following command:
+
+```shell
+helm install ngf oci://ghcr.io/nginx/charts/nginx-gateway-fabric \
+  --create-namespace -n nginx-gateway \
+  --set nginxAgent.enable=true \
+  --set nginxAgent.dataplaneKeySecretName=<data_plane_key_secret_name> \
+  --set nginxAgent.endpointHost=agent.connect.nginx.com
```

Comment:
@bjee19 @sjberman is this what you're looking for? (Also for current lines 93-95)

```suggestion
helm install ngf oci://ghcr.io/nginx/charts/nginx-gateway-fabric \
  --create-namespace -n nginx-gateway \
  --set nginx.nginxOneConsole.enable=true \
  --set nginx.nginxOneConsole.dataplaneKeySecretName=<data_plane_key_secret_name> \
  --set nginx.nginxOneConsole.endpointHost=agent.connect.nginx.com
```

Reply:
The `enable` field isn't necessary, that is not an option in NGF. Also, the `endpointHost` value is already a default, so I don't know that we need to show that in this example. Probably just need the dataplaneKeySecretName value

Reply:
So maybe

```suggestion
helm install ngf oci://ghcr.io/nginx/charts/nginx-gateway-fabric \
  --create-namespace -n nginx-gateway \
  --set nginx.nginxOneConsol.dataplaneKeySecretName=<data_plane_key_secret_name> \
```

Reply:
`--create-namespace` isn't really necessary because it will have to exist before installation anyway to add the dataplane key secret to.

Reply:
Thus we'll end up with:

```
helm install ngf oci://ghcr.io/nginx/charts/nginx-gateway-fabric \
  --set nginx.nginxOneConsole.dataplaneKeySecretName=<data_plane_key_secret_name>
```

Reply:
Like this? 

```
helm install ngf oci://ghcr.io/nginx/charts/nginx-gateway-fabric \
  --set nginx.nginxOneConsole.dataplaneKeySecretName=<data_plane_key_secret_name> \
 -n nginx-gateway
```

Reply:
Yes, that's good.